Introduce volume correction

Robert-J-H opened the pull request:
IAudioEndpointVolume is now used to read and set mute state and levels.
The sending of volume up/down is kept as fall-back procedure.
Should for some reason the correction not work,so is it frequently
amendable by reloading addons (Control+NVDA+F3).
The GUI does currently not reflect the enhanced possibilities.

Robert-J-H commented:
Hi Noelia

Thanks for merging.

You've merged all three commits, right?
In this case, you should check if the functionality is now better.
I've re-enabled the volume settings for all profiles.
As you might remember, I was wondering why I didn't see them occasionally...
Anyway, it's back in but it is only written to the normal configuration.
I hope this is OK. I think it would not make sense to alter the master
volume on each profile switching.
I've also added my name and tweaked the description a bit.
There's still work to be done:
- changing the volume check box to a choice combo (same as for
NumLock) with three entries.
-Off
- minimum
- absolute
(I have to ponder about the exact wording).
I also want to add an event handler to the spin box such that a sample
text is spoken at the level of the chosen setting when you arrow up
and down.
Not much text, perhaps just "Percent".
I wonder if we should set a lower limit if "absolute" is selected. It
is bad if the user can set the volume to 0 and he hasn't a Braille
display.... ;-)

Cheers
Robert

On 01/02/2018, nvdaes <notifications@github.com> wrote:
> Merged #1.
>
> --
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/nvdaes/numLockManager/pull/1#event-1452870868

nvdaes commented:
I agree. I had also thought that setting volume to 0 isn't good.
If you want, make your changes and then I can review them and provide
a better configuration key for numLock (not just activation).
Cheers

2018-02-01 14:58 GMT+01:00, Robert Haenggi <notifications@github.com>:
> Hi Noelia
>
> Thanks for merging.
>
> You've merged all three commits, right?
> In this case, you should check if the functionality is now better.
> I've re-enabled the volume settings for all profiles.
> As you might remember, I was wondering why I didn't see them
> occasionally...
> Anyway, it's back in but it is only written to the normal configuration.
> I hope this is OK. I think it would not make sense to alter the master
> volume on each profile switching.
> I've also added my name and tweaked the description a bit.
> There's still work to be done:
> - changing the volume check box to a choice combo (same as for
> NumLock) with three entries.
> -Off
> - minimum
> - absolute
> (I have to ponder about the exact wording).
> I also want to add an event handler to the spin box such that a sample
> text is spoken at the level of the chosen setting when you arrow up
> and down.
> Not much text, perhaps just "Percent".
> I wonder if we should set a lower limit if "absolute" is selected. It
> is bad if the user can set the volume to 0 and he hasn't a Braille
> display.... ;-)
>
> Cheers
> Robert
>
> On 01/02/2018, nvdaes <notifications@github.com> wrote:
>> Merged #1.
>>
>> --
>> You are receiving this because you authored the thread.
>> Reply to this email directly or view it on GitHub:
>> https://github.com/nvdaes/numLockManager/pull/1#event-1452870868
>
>
> --
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub:
> https://github.com/nvdaes/numLockManager/pull/1#issuecomment-362273286

Robert-J-H commented:
Hi Noelia
This whole configuration business gives me some headaches...
I've changed the "changeVolumeLevel" key from boolean to integer.
However, this gives a validation error and I've set up some code to
transform boolean to integer.
In order to avoid checking all the time, I've also written a short
installTasks.py file.
I think you could place there some cleanup code as well if you want to
change the configuration keys.
I'm currently facing the problem that the volume level is not updated
on reloading add-ons (NVDA+Control+F3).
I have to investigate how I can fool the program to avoid the cache.

Cheers
Robert

On 01/02/2018, nvdaes <notifications@github.com> wrote:
> I agree. I had also thought that setting volume to 0 isn't good.
> If you want, make your changes and then I can review them and provide
> a better configuration key for numLock (not just activation).
> Cheers
>
> 2018-02-01 14:58 GMT+01:00, Robert Haenggi <notifications@github.com>:
>> Hi Noelia
>>
>> Thanks for merging.
>>
>> You've merged all three commits, right?
>> In this case, you should check if the functionality is now better.
>> I've re-enabled the volume settings for all profiles.
>> As you might remember, I was wondering why I didn't see them
>> occasionally...
>> Anyway, it's back in but it is only written to the normal configuration.
>> I hope this is OK. I think it would not make sense to alter the master
>> volume on each profile switching.
>> I've also added my name and tweaked the description a bit.
>> There's still work to be done:
>> - changing the volume check box to a choice combo (same as for
>> NumLock) with three entries.
>> -Off
>> - minimum
>> - absolute
>> (I have to ponder about the exact wording).
>> I also want to add an event handler to the spin box such that a sample
>> text is spoken at the level of the chosen setting when you arrow up
>> and down.
>> Not much text, perhaps just "Percent".
>> I wonder if we should set a lower limit if "absolute" is selected. It
>> is bad if the user can set the volume to 0 and he hasn't a Braille
>> display.... ;-)
>>
>> Cheers
>> Robert
>>
>> On 01/02/2018, nvdaes <notifications@github.com> wrote:
>>> Merged #1.
>>>
>>> --
>>> You are receiving this because you authored the thread.
>>> Reply to this email directly or view it on GitHub:
>>> https://github.com/nvdaes/numLockManager/pull/1#event-1452870868
>>
>>
>> --
>> You are receiving this because you modified the open/close state.
>> Reply to this email directly or view it on GitHub:
>> https://github.com/nvdaes/numLockManager/pull/1#issuecomment-362273286
>
>
> --
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/nvdaes/numLockManager/pull/1#issuecomment-362276804
